src/megaclient.cpp

```diff
@@ -7601,10 +7627,221 @@ void MegaClient::fetchnodes()
         }
 #endif
 
+        fetchkeys();
         reqs.add(new CommandFetchNodes(this));
     }
 }
 
+void MegaClient::fetchkeys()
+{
+#ifdef ENABLE_CHAT
+
+    fetchingkeys = true;
+
+    User *u = finduser(me, 1);
+
+    int creqtag = reqtag;
+    reqtag = 0;
+    reqs.add(new CommandPubKeyRequest(this, u));    // public RSA
+    reqtag = creqtag;
+
+    getua(u, "*keyring", 0);   // private Cu25519 & private Ed25519
+    getua(u, "+puEd255", 0);
+    getua(u, "+puCu255", 0);
+    getua(u, "+sigCu255", 0);
+    getua(u, "+sigPubk", 0); // it triggers MegaClient::initializekeys()
```

Comment:
As long as there is response to the request of `+sigPubk`, it will call `initializekeys()`. It doesn't really matter the response for the other attributes. Also, it doesn't matter if `+sigPubk` is found or not, it will trigger the initialization.
What's the code path you are concerned? I only see when an `API_EINTERNAL` is detected.